Fail building on SmartOS

Seems related: https://github.com/joyent/libuv/issues/1195 fixed by https://github.com/joyent/libuv/commit/72337966d4a7cdd32b990b9618c4b2349c18042b

But I'm not sure where to put `-D_XOPEN_SOURCE=500` to workaround this?

```console
[root@swarmcc /swarmcc/edh-universe/haskell/network]# cabal build network
Resolving dependencies...
Build profile: -w ghc-8.8.4 -O1
In order, the following will be built (use -v for more details):
 - network-3.1.2.2 (lib:network) (first run)
Configuring network-3.1.2.2...
configure: WARNING: unrecognized options: --with-compiler
checking build system type... x86_64-pc-solaris2.11
checking host system type... x86_64-pc-solaris2.11
checking for gcc... /opt/local/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /opt/local/bin/gcc accepts -g... yes
checking for /opt/local/bin/gcc option to accept ISO C89... none needed
checking for an ANSI C-conforming const... yes
checking how to run the C preprocessor... /opt/local/bin/gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/types.h... (cached) yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/uio.h usability... yes
checking sys/uio.h presence... yes
checking for sys/uio.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking netinet/tcp.h usability... yes
checking netinet/tcp.h presence... yes
checking for netinet/tcp.h... yes
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking net/if.h usability... yes
checking net/if.h presence... yes
checking for net/if.h... yes
checking netioapi.h usability... no
checking netioapi.h presence... no
checking for netioapi.h... no
checking for struct ucred... no
checking for gai_strerror... no
checking for gethostent... no
checking for accept4... no
checking for getpeereid... no
checking whether AI_ADDRCONFIG is declared... yes
checking whether AI_ALL is declared... yes
checking whether AI_NUMERICSERV is declared... yes
checking whether AI_V4MAPPED is declared... yes
checking whether IPV6_V6ONLY is declared... yes
checking whether IPPROTO_IP is declared... yes
checking whether IPPROTO_TCP is declared... yes
checking whether IPPROTO_IPV6 is declared... yes
checking whether SO_PEERCRED is declared... no
checking for struct msghdr.msg_control... no
checking for struct msghdr.msg_accrights... yes
checking for struct sockaddr.sa_len... no
configure: creating ./network.buildinfo
configure: creating ./config.status
config.status: creating include/HsNetworkConfig.h
configure: WARNING: unrecognized options: --with-compiler
Preprocessing library for network-3.1.2.2..
compiling /swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/Network/Socket/Posix/MsgHdr_hsc_make.c failed (exit code 1)
rsp file was: "/swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/Network/Socket/Posix/hsc2hscall30540-0.rsp"
command was: /opt/local/bin/gcc -c /swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/Network/Socket/Posix/MsgHdr_hsc_make.c -o /swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/Network/Socket/Posix/MsgHdr_hsc_make.o -m64 -m64 -L/opt/local/gcc9/lib/gcc/x86_64-sun-solaris2.11/9.3.0 -Wl,-R/opt/local/gcc9/lib/gcc/x86_64-sun-solaris2.11/9.3.0 -L/opt/local/lib -L/usr/lib/amd64 -Wl,-R/usr/lib/amd64 -Wl,-R/opt/local/lib -m64 -D__GLASGOW_HASKELL__=808 -Dsolaris2_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Dsolaris2_HOST_OS=1 -Dx86_64_HOST_ARCH=1 -Iinclude -I/swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/include -I/swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/autogen -I/swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/global-autogen -include /swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/autogen/cabal_macros.h -I/opt/local/lib/ghc-8.8.4/unix-2.7.2.2/include -I/opt/local/lib/ghc-8.8.4/time-1.9.3/include -I/opt/local/lib/ghc-8.8.4/bytestring-0.10.10.1/include -I/opt/local/include -I/opt/local/lib/ghc-8.8.4/base-4.13.0.0/include -I/opt/local/include -I/opt/local/lib/ghc-8.8.4/integer-gmp-1.0.2.0/include -I/opt/local/lib/ghc-8.8.4/include -I/opt/local/include -I/opt/local/lib/ghc-8.8.4/include/
error: In file included from /opt/local/lib/ghc-8.8.4/template-hsc.h:11,
                 from /swarmcc/edh-universe/dist-newstyle/build/x86_64-solaris/ghc-8.8.4/network-3.1.2.2/build/Network/Socket/Posix/MsgHdr_hsc_make.c:1:
MsgHdr.hsc: In function 'main':
/opt/local/lib/ghc-8.8.4/template-hsc.h:82:24: error: 'struct msghdr' has no member named 'msg_control'
   82 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:35:5: note: in expansion of macro 'hsc_peek'
/opt/local/lib/ghc-8.8.4/template-hsc.h:82:24: error: 'struct msghdr' has no member named 'msg_controllen'
   82 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:36:5: note: in expansion of macro 'hsc_peek'
/opt/local/lib/ghc-8.8.4/template-hsc.h:82:24: error: 'struct msghdr' has no member named 'msg_flags'
   82 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:37:5: note: in expansion of macro 'hsc_peek'
/opt/local/lib/ghc-8.8.4/template-hsc.h:86:24: error: 'struct msghdr' has no member named 'msg_control'
   86 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:49:5: note: in expansion of macro 'hsc_poke'
/opt/local/lib/ghc-8.8.4/template-hsc.h:86:24: error: 'struct msghdr' has no member named 'msg_controllen'
   86 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:50:5: note: in expansion of macro 'hsc_poke'
/opt/local/lib/ghc-8.8.4/template-hsc.h:86:24: error: 'struct msghdr' has no member named 'msg_flags'
   86 |                 (long) offsetof (t, f));
      |                        ^~~~~~~~
MsgHdr.hsc:51:5: note: in expansion of macro 'hsc_poke'

[root@swarmcc /swarmcc/edh-universe/haskell/network]#
```

--- a/network.cabal
+++ b/network.cabal
@@ -111,6 +111,8 @@ library
 
   if os(solaris)
     extra-libraries: nsl, socket
+    cpp-options: -D__EXTENSIONS__ -D_XOPEN_SOURCE=500
+    cc-options: -D__EXTENSIONS__ -D_XOPEN_SOURCE=500
 
   if os(windows)
     other-modules:

